exasol-s3/src/main/java/com/exasol/spark/s3/ExasolWriteBuilderProvider.java

```diff
@@ -44,7 +48,23 @@ public ExasolWriteBuilderProvider(final ExasolOptions options) {
      */
     public WriteBuilder createWriteBuilder(final StructType schema, final LogicalWriteInfo defaultInfo) {
         final SparkSession sparkSession = SparkSession.active();
-        final LogicalWriteInfo info = getUpdatedLogicalWriteInfo(defaultInfo, sparkSession);
+        final String s3Bucket = this.options.getS3Bucket();
+        final String s3BucketKey = this.s3BucketKeyPathProvider.getS3BucketKeyForWriteLocation(defaultInfo.queryId());
+        validateWritePathIsEmpty(s3Bucket, s3BucketKey);
+        final LogicalWriteInfo info = getUpdatedLogicalWriteInfo(defaultInfo, s3Bucket, s3BucketKey, sparkSession);
+        return createCSVWriteBuilder(sparkSession, schema, info);
+    }
+
+    private void validateWritePathIsEmpty(final String s3Bucket, final String s3BucketKey) {
+        try (final S3FileSystem s3FileSystem = S3FileSystem.fromOptions(this.options)) {
+            if (!s3FileSystem.isEmpty(s3Bucket, Optional.of(s3BucketKey))) {
+                throw new ExasolValidationException("bucketIs not empty");
```

Comment:
Use error reporting if possible and add bucket name and key to the error message

Reply:
Opps, forgot to update this. Going to add error, message and code.